src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/transformation/services/TransformationInputToCaseTransformer.java

```diff
@@ -43,17 +47,30 @@ public TransformationInputToCaseTransformer(
     // endregion
 
     public SuccessfulTransformationResponse toCase(TransformationInput transformationInput) {
-        transformationInputValidator.assertIsValid(transformationInput);
+        List<String> errors = transformationInputValidator.getErrors(transformationInput);
```

Comment:
you can keep this error as it is now so if there are errors it should fail for both automated and normal case creation.
Next you can check for warnings and if it is automated and warnings are present throw Exception with HttpStatus.UNPROCESSABLE_ENTITY so that transformation API would fallback to exception record creation

Reply:
The point in this change is to combine errors which are errors in both cases (automated and not automated) with warnings which are errors only if it is automated. If we keep it as it is (i.e. throwing exception straightaway if errors are found) warnings will be missing as errors if it is automated.

Reply:
I don't think this is valid. I think that was one of the discussion as we had not gone through the code at that point. Combining errors and warnings is bit a of a hack to be honest. As per transformation contract you should throw 422 which means structure is valid but data cannot be used to create the case.